fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1808,20 +1808,57 @@ private static boolean checkFirstKeyIsHeader(@Nonnull KeyValue firstKeyValue,
                                                  @Nonnull SubspaceProvider subspaceProvider,
                                                  @Nonnull Subspace subspace,
                                                  @Nonnull StoreExistenceCheck existenceCheck) {
-        if (TupleHelpers.equals(subspace.unpack(firstKeyValue.getKey()), Tuple.from(STORE_INFO_KEY))) {
+        final Tuple firstKey = subspace.unpack(firstKeyValue.getKey());
+        if (TupleHelpers.equals(firstKey, Tuple.from(STORE_INFO_KEY))) {
             return true;
         } else if (existenceCheck == StoreExistenceCheck.NONE) {
             LOGGER.warn(KeyValueLogMessage.of("Record store has no info but is not empty",
                     subspaceProvider.logKey(), subspaceProvider.toString(context),
-                    LogMessageKeys.KEY, subspace.unpack(firstKeyValue.getKey())));
+                    LogMessageKeys.KEY, firstKey));
             return false;
+        } else if (existenceCheck == StoreExistenceCheck.ERROR_IF_NO_INFO_AND_HAS_RECORDS_OR_INDEXES) {
+            final FDBRecordStoreKeyspace keyspace = determineRecordStoreKeyspace(firstKey, subspaceProvider, context);
+            // White list of acceptable key ranges for the first key. This may need to be updated as more keyspaces are added.
+            if (FDBRecordStoreKeyspace.INDEX_STATE_SPACE.equals(keyspace) || FDBRecordStoreKeyspace.INDEX_RANGE_SPACE.equals(keyspace)) {
```

Comment:
I guess you have a good reason not excluding `RECORD_COUNT`, `INDEX_UNIQUENESS_VIOLATIONS_SPACE` or `RECORD_VERSION_SPACE`, which worth to be documented here.

Reply:
Added additional comments about why I chose what I chose.